crates/librqbit/src/blocklist.rs

```diff
@@ -0,0 +1,339 @@
+use anyhow::Result;
+use async_compression::tokio::bufread::GzipDecoder;
+use futures::TryStreamExt;
+use intervaltree::IntervalTree;
+use std::net::IpAddr;
+use std::pin::Pin;
+use std::str::FromStr;
+use tokio::io::AsyncRead;
+use tokio::{io::AsyncBufReadExt, io::BufReader};
+use tokio_util::io::StreamReader;
+use tracing::{debug, info, trace};
+
+pub struct Blocklist {
+    // Separate trees for IPv4 and IPv6 since they have different numeric ranges
+    ipv4_ranges: IntervalTree<u32, ()>,
```

Comment:
I think you can store Ipv4Addr/Ipv6Addr everywhere here as they implement Ord, and that's the only necessary thing for both std::ops::Range and IntervalRange.